workbase/src/renderer/components/SchemaDesign/store/actions.js

```diff
@@ -189,6 +185,22 @@ export default {
     await dispatch(COMMIT_TX, graknTx)
       .then(async () => {
         await dispatch(UPDATE_METATYPE_INSTANCES);
+
+        graknTx = await dispatch(OPEN_GRAKN_TX);
+
+        const type = await graknTx.getSchemaConcept(payload.attributeLabel);
+
+        Object.assign(type, { label: payload.attributeLabel });
+
+        const data = await buildSchema([type]);
+
+        data.edges.push(...((await typeInboundEdges(type, state.visFacade)).filter(x => x.to === type.id)));
+
+        state.visFacade.addToCanvas({ nodes: data.nodes, edges: data.edges });
+
+        data.nodes = await computeAttributes(data.nodes);
+        state.visFacade.updateNode(data.nodes);
+        graknTx.close();
```

Comment:
what's going on here? can you make it a method or add comments? so that in the future we remember